source/change-streams/tests/README.rst

```diff
@@ -164,43 +164,28 @@ The following tests have not yet been automated, but MUST still be tested
       - ``getResumeToken`` must return the ``_id`` of the last document returned if one exists.
       - ``getResumeToken`` must return ``resumeAfter`` from the initial aggregate if the option was specified.
       - If ``resumeAfter`` was not specified, the ``getResumeToken`` result must be empty.
 13. - For a ``ChangeStream`` under these conditions:
       - The batch is not empty.
       - The batch has been iterated up to but not including the last element.
-   - Expected result:
+    - Expected result:
       - ``getResumeToken`` must return the ``_id`` of the previous document returned.
 14. - For a ``ChangeStream`` under these conditions:
       - The batch is not empty.
       - The batch hasn’t been iterated at all.
       - Only the initial ``aggregate`` command has been executed.
-   - Expected result:
+    - Expected result:
       - ``getResumeToken`` must return ``startAfter`` from the initial aggregate if the option was specified.
       - ``getResumeToken`` must return ``resumeAfter`` from the initial aggregate if the option was specified.
       - If neither the ``startAfter`` nor ``resumeAfter`` options were specified, the ``getResumeToken`` result must be empty.
-15. - For a ``ChangeStream`` under these conditions:
-      - Running against a server ``>=4.0.7``.
-      - The batch is not empty.
-      - The batch hasn’t been iterated at all.
-      - The stream has iterated beyond a previous batch and a ``getMore`` command has just been executed.
-   - Expected result:
-      - ``getResumeToken`` must return the ``postBatchResumeToken`` from the previous command response.
-16. - For a ``ChangeStream`` under these conditions:
-      - Running against a server ``<4.0.7``.
-      - The batch is not empty.
-      - The batch hasn’t been iterated at all.
-      - The stream has iterated beyond a previous batch and a ``getMore`` command has just been executed.
-   - Expected result:
-      - ``getResumeToken`` must return the ``_id`` of the previous document returned if one exists.
-      - ``getResumeToken`` must return ``resumeAfter`` from the initial aggregate if the option was specified.
-      - If neither the ``startAfter`` nor ``resumeAfter`` options were specified, the ``getResumeToken`` result must be empty.
-#. ``$changeStream`` stage for ``ChangeStream`` started with ``startAfter`` against a server ``>=4.1.1`` that has not received any results yet MUST include a ``startAfter`` option and MUST NOT include a ``resumeAfter`` option when resuming a change stream.
-#. ``$changeStream`` stage for ``ChangeStream`` started with ``startAfter`` against a server ``>=4.1.1`` that has received at least one result MUST include a ``resumeAfter`` option and MUST NOT include a ``startAfter`` option when resuming a change stream.
+    - Note that this test cannot be run against sharded topologies because in that case the initial ``aggregate`` command only establishes cursors on the shards and always returns an empty ``firstBatch``.
```

Comment:
I think you also mentioned that all of the current spec tests are only expected to run against replica sets. I can also verify that by the `topology` field in each test.

Was this the only prose test that cannot be run against a sharded cluster? If so, perhaps we should mention earlier in the document that other prose tests _should_ be run against replica set and sharded clusters.

I also saw your discussion in `#server-query` and the answers explaining why sharded clusters are a unique case where we need to assert that a change eventually appears but cannot expect it in a particular batch. It sounds like there's an action item to revisit/rethink our driver testing for that, so would you mind creating a separate spec ticket to summarize the discussion/todos so we don't lose track of it?

Reply:
Opened https://jira.mongodb.org/browse/SPEC-1374 and https://jira.mongodb.org/browse/SPEC-1375.
I will also update the prose tests to run the tests on sharded clusters.